src/index.ts

```diff
@@ -286,22 +274,18 @@ function configToMatchingEvents(config: SalesforcePluginConfig): string[] {
     return []
 }
 
-export async function onEvent(event: PluginEvent, { global, config }: SalesforcePluginMeta): Promise<void> {
-    if (!global.buffer) {
-        throw new Error(`There is no buffer. Setup must have failed, cannot process event: ${event.event}`)
-    }
-
+export function shouldSendEvent(event: PluginEvent, meta: SalesforcePluginMeta): boolean {
+    const { config } = meta
     const eventsToMatch = configToMatchingEvents(config)
-    if (!eventsToMatch.includes(event.event)) {
+    return eventsToMatch.includes(event.event)
+}
+
+export async function onEvent(event: PluginEvent, meta: SalesforcePluginMeta): Promise<void> {
+    if (!shouldSendEvent(event, meta)) {
         return
     }
 
-    const eventSize = JSON.stringify(event).length
-    global.buffer.add(event, eventSize)
-}
-
-export function teardownPlugin({ global }: SalesforcePluginMeta): void {
-    global.buffer.flush()
+    await sendEventToSalesforce(event, meta, await getToken(meta))
```

Comment:
I guess we `await` sends in most other plugins, right? Technically the buffer would do sends in the background and not block up the pipeline (I think?). Just wanted to point out the small change in behavior here.

Reply:
yes exactly, this is what we do in all other plugins